File: cpp/Array/L_findTheStdReplaceChalk.cpp
int chalkReplacer(vector<int> &chalk, int k)
{
    const int n = chalk.size();
    vector<long long> sum(n, chalk[0]); // prefix sum 0-indexed
    for (int i = 1; i < n; i++)
    {
        sum[i] = sum[i - 1] + chalk[i];
        //    cout<<i<<"->"<<sum[i]<<endl;
    }
    k %= sum[n - 1];
    //    cout<<"k="<<k<<endl;

    return upper_bound(sum.begin(), sum.end(), k) - sum.begin();
}

File: JavaScript/Array/L_findTheStdReplaceChalk.js
var chalkReplacer = function (chalk, k) {
    let sum = 0;
    for (let x of chalk) sum += x;
    let rem = k % sum;
    for (let i = 0; i < chalk.length; i++) {
        rem -= chalk[i];
        if (rem < 0) return i;
    }
};
